### PR TITLE
feat: context config

### PR DESCRIPTION
**Pull request by miraclx**
Add context configuration contract

We decided to temporarily store strongly consistent data like application versions & member lists on-chain while we focus on eventual consistency for private data.

Features:

- Anybody can create a context so far as they can prove ownership of the private key that derived public key represented as the context ID.
	- This means signing the context creation payload. This payload includes a timestamp to mitigate replay attacks, the permitted threshold is configurable only by the contract account.
- Granular permissions, accounts can granted permissions to update the application or manage the member list (invite & remove).
- Paginated member list fetching.

### CODE REVIEW

**Comment by xilosada** [comment]
missing test.sh

File: contracts/context-config/src/repr.rs [hunk 1]
```diff
@@ -0,0 +1,145 @@
+use std::{fmt, ops::Deref};
+
+use near_sdk::{bs58, near};
+use thiserror::Error;
+
+#[derive(Eq, Ord, Copy, Clone, Debug, PartialEq, PartialOrd)]
+#[near(serializers = [borsh, json])]
+#[serde(transparent)]
+pub struct Repr<T> {
```
Comment (MatejVukosav) [inline contracts/context-config/src/repr.rs:13]: What does this mean? Is this "Representation"?

Comment (miraclx) [reply contracts/context-config/src/repr.rs:13]: `Repr` makes generic bytes representable over the wire, encoding them as `bs58`.

**Comment by miraclx** [comment]
> missing test.sh

the one we have for package manager is broken, and has been for months, so I guess nobody uses it

but anyway, `cargo test` is all that's needed here, not sure a script that just calls `cargo test` is that useful, wdyt?

File: contracts/context-config/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+mod types;
+
+pub use repr::{Repr, ReprBytes};
+pub use types::{
+    ApplicationId, ApplicationSource, BlobId, ContextId, ContextIdentity, Guard, SignedPayload,
+};
+
+#[derive(Debug)]
+#[near(contract_state)]
+pub struct ContextConfig {
```
Comment (MatejVukosav) [inline contracts/context-config/src/lib.rs]: Structs hierarchy is confusing. Lets discuss it today

**Comment by xilosada** [comment]
@miraclx then let's remove the file (in another PR) and ensure the tests run in the ci

File: contracts/context-config/src/lib.rs (hunk 2 above)
Comment (MatejVukosav) [reply contracts/context-config/src/lib.rs]: `ContextsConfigs` or something similar to indicate more configurations

**Comment by miraclx** [comment]
> and ensure the tests run in the ci

the tests are running

**Review by MatejVukosav — APPROVED** [review]

**Pull request merged**